### ISSUE TITLE
Key identifier?

### ISSUE THREAD

**Issue by quartzjer** [tplooker/draft-looker-bls-jose-cose#12]
I'm posing this as a question and not a direct suggestion.

The key identifiers are specifically for only one subgroup (the `G1` on `Bls48581G1 ` for example) on the defined pairing curve, and in practice an implementation will use that single group identifier to first identify the pairing curve itself, caring only about the `Bls48581` prefix.  Would it make sense to add a separator character such as `_`, `-`, or `+` like in other JOSE/COSE identifiers?

In fact, one could argue that `Bls48581` is technically the curve name and the group could be identified via a separate element, but that's being rather pedantic :)

**Comment by tplooker** [comment]
> The key identifiers are specifically for only one subgroup (the G1 on Bls48581G1 for example) on the defined pairing curve, and in practice an implementation will use that single group identifier to first identify the pairing curve itself, caring only about the Bls48581 prefix. Would it make sense to add a separator character such as _, -, or + like in other JOSE/COSE identifiers?

Yes this could be a good suggestion if we believe implementations will benefit from being able to partially parse the `crv` value to know the underlying curve involved is and then further process OR use another field to establish what sub-group is being used. Im unsure however how common that behaviour may be?

**Comment by selfissued** [comment]
@tplooker wrote:
> Yes this could be a good suggestion if we believe implementations will benefit from being able to partially parse the crv value to know the underlying curve involved is and then further process OR use another field to establish what sub-group is being used. Im unsure however how common that behaviour may be?

I'm not in favor of partially parsing identifiers.  We haven't done that at any other place in COSE or JOSE, to my knowledge, even though they sometimes do have human-comprehensible structure, such as "A128CBC-HS256".

I'm in favor of closing this issue on that basis.
